### PR TITLE
[rayci] add windows job support

### PR DESCRIPTION
**Pull request by aslonnie**
also splitting the long convert function into smaller ones.

most of the step logic is shared. the main difference is how the docker plugin field is inserted.

### CODE REVIEW

File: raycicmd/converter.go [hunk 1]
```diff
@@ -110,71 +112,68 @@ func parseStepEnvs(v any) ([]*envEntry, error) {
...
-func (c *converter) convertPipelineStep(step map[string]any) (
-	map[string]any, error,
-) {
-	if _, ok := step["wait"]; ok {
-		// a wait step
-		if err := checkStepKeys(step, waitStepAllowedKeys); err != nil {
-			return nil, fmt.Errorf("check wait step keys: %w", err)
-		}
-		return cloneMapExcept(step, waitStepDropKeys), nil
+func (c *converter) convertWait(step map[string]any) (map[string]any, error) {
```
Comment (aslonnie) [inline raycicmd/converter.go:115]: this is just splitting the function out

File: raycicmd/converter.go [hunk 2]
```diff
@@ -110,71 +112,68 @@ func parseStepEnvs(v any) ([]*envEntry, error) {
...
-		label, _ := stringInMap(step, "label")
-		instanceType, _ := stringInMap(step, "instance_type")
+	return cloneMapExcept(step, waitStepDropKeys), nil
+}
 
-		var matrix any
-		if m, ok := step["matrix"]; ok {
-			matrix = m
-		}
+func (c *converter) convertWanda(step map[string]any) (map[string]any, error) {
```
Comment (aslonnie) [inline raycicmd/converter.go:123]: this is also just splitting the wanda conversion logic out into its own function.

File: raycicmd/bk_pipeline.go [hunk 3]
```diff
@@ -56,14 +56,38 @@ type stepDockerPluginConfig struct {
...
 	envs := append([]string(nil), buildkiteEnvs...)
 	if len(config.extraEnvs) > 0 {
 		envs = append(envs, config.extraEnvs...)
 	}
 
+	m := map[string]any{
+		"image":          windowsBuildEnvImage,
+		"shell":          []string{"bash", "-c"},
+		"shm-size":       "2.5gb",
+		"mount-checkout": true,
```
Comment (can-anyscale) [inline raycicmd/bk_pipeline.go:79]: does this mount the checkout to `/ray` workdir?; also can we amount the docker.sock as well for consistency? the artifact-mount can probably be done at a later time.

**Review by can-anyscale — APPROVED** [review]

File: .buildkite/windows.rayci.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+group: windows
+steps:
+  - name: ":windows: windows example"
+    job_env: WINDOWS
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply .buildkite/windows.rayci.yaml:4]: this is not really an image name, but a special builtin case handled by rayci.

File: raycicmd/bk_pipeline.go (hunk 3 above)
Comment (aslonnie) [reply raycicmd/bk_pipeline.go:79]: it checks out at a temp dir, and mount it into `c:\workdir` inside the image.

the docker sock probably does not work.. the socket is a "unix domain socket", and my educated guess is that it won't work on windows. google search says it needs to mount in the pipe.

I am trying to keep the status quo of how windows run on civ1 for now, and get it migrated first, and then we can iterate from there.

**Pull request merged**